torch/csrc/jit/ir.cpp

```diff
@@ -251,6 +256,335 @@ std::ostream& operator<<(std::ostream & out, const Graph & g) {
+    const size_t level
+  ) {
+    // if there are subblocks on this node, visit them
+    switch (node->kind()) {
+    case prim::Return:
+      // Handled elsewhere, do nothing
+      break;
+    case prim::Constant:
+      break;
+    case prim::Loop:
+      printLoop(out, node, level);
+      break;
+    case prim::If:
+      printIf(out, node, level);
+      break;
+    default:
+      indent(out, level);
+      // Print outputs
+      if (node->outputs().size() > 0) {
+        auto delim = "";
+        for (const auto* output_value : node->outputs()) {
+          out << delim;
+          printValue(out, output_value);
+          delim = ", ";
+        }
+        out << " = ";
+      }
+
+      out << node->kind().toQualString();
+
+      // Print instruction parameters
+      printValueList(out, node->inputs());
+
+      out << std::endl;
+    }
+
+    return out;
+  }
+
+  std::ostream& printReturn(
+    std::ostream & out,
+    const Node* node,
+    const size_t level
+  ) {
+    indent(out, level);
+    const auto& returns = node->inputs();
+    if (returns.size() > 0) {
+      out << "return ";
+      std::string delimiter = "";
+      if (returns.size() > 1) {
+        printValueList(out, returns);
+      } else {
+        printValue(out, returns[0]);
+      }
+      out << std::endl;
+    }
+    return out;
+  }
+
+  std::ostream& printBlock(
+    std::ostream & out,
+    const Block* root,
+    const size_t level
+  ) {
+    for (const auto* node : root->nodes()) {
+      printNode(out, node, level);
+    }
+
+    printNode(out, root->return_node(), level);
+
+    return out;
+  }
+
+  inline bool isNameUnique(std::string& name, const Value* val) const {
+    auto generated_name_value = generated_names_.find(name);
+    if (generated_name_value != generated_names_.end() &&
+        generated_name_value->second != val) {
+      // Found a generated name match, check that it's for a different value
+      return false;
+    }
+    return graph_.uniqueNames().find(name) == graph_.uniqueNames().end();
+  }
+
+  std::ostream& printValue(
+    std::ostream & out,
+    const Value* val
+  ) {
+    const auto node = val->node();
+    if (node->kind() == prim::Constant) {
+      printAttributeValue(out, node->attributeNames()[0], node);
+      return out;
+    }
+
+    auto name_source = val;
+
+    auto aliased_name = aliases_.find(val);
+    if (aliased_name != aliases_.end()) {
+      name_source = aliased_name->second;
+    }
+
+    auto name = name_source->uniqueName();
+
+    bool using_generated_name = false;
+    if (isdigit(name.at(0))) {
+      std::stringstream ss;
+      ss << "t" << name;
+      name = ss.str();
+      using_generated_name = true;
+    } else if (name.find_last_of('.') != std::string::npos) {
+      // Make unique name a valid variable name (e.g. a.1 -> a1)
+      name.erase(std::remove(name.begin(), name.end(), '.'), name.end());
+      using_generated_name = true;
+    }
+
+    if (using_generated_name) {
```

Comment:
Isn't this going to run every time a Value is referenced? I feel the `Value* -> pretty_name` mapping should be cached.